test/check-session

```diff
@@ -28,25 +28,27 @@ class TestSession(MachineCase):
         # TODO: Make this work with non-root
         b.default_user = "root"
 
-        def check_session(should_exist):
-            check = check_in if should_exist else check_not_in
-            check(m.execute("loginctl list-sessions"), "admin")
+        def wait_session(should_exist):
+            def cond():
+                check = check_in if should_exist else check_not_in
+                return check(m.execute("loginctl list-sessions"), "admin")
+            wait (cond)
 
         def kill_user_admin():
             # logind from systemd 208 is buggy, so we use systemd directly if it fails
             # https://bugs.freedesktop.org/show_bug.cgi?id=71092
             m.execute("loginctl kill-user admin || systemctl kill user-1000.slice")
 
-        check_session(False)
+        wait_session(False)
```

Comment:
Technically it doesn't hurt to wait here, but in theory a check should be enough (we didn't kill anything).
What do you think? Consistently use wait or an actual check when we think the state should already be reached.